src/Controllers/SedimentController.ts

```diff
@@ -119,6 +128,29 @@ module StreamStats.Controllers {
             //     }
             // }
         }
+        private verifyExtensionCanContinue() {
```

Comment:
I was able to enter this invalid date range by typing it in and tab-ing over to the Ok button:

![image](https://github.com/USGS-WiM/StreamStats/assets/40237491/5e1e7540-014c-45ee-a36e-b28f5da7407e)

The Ok button just blinked and didn't give a success message. I got this console error:

![image](https://github.com/USGS-WiM/StreamStats/assets/40237491/80c2b749-07cb-427e-880d-c62a9f90e268)

Maybe we should do some additional error checking to make sure startDate and endDate are defined?

Reply:
> a simple fix could be that I just check that that is defined and if not, still return the same "Date range is not valid." error

That sounds good to me

Reply:
It may be as simple as checking `if (this.dateRange.dates) {` rather than just `if (this.dateRange) {` which it is currently doing as the very first check

Reply:
Maybe `if (this.dateRange.dates.startDate && this.dateRange.dates.endDate) {`

Reply:
ok, attempt made to address the last review